### ISSUE TITLE
Tests fail while using collection caching of partials

### ISSUE THREAD

**Issue by prathamesh-sonpatki** [rails/rails#24132]
I am trying to setup collection caching of partials here - https://github.com/codetriage/codetriage/pull/454/files but getting test failures as follows - 

```
Error:
MaintainingRepoSubscriptionsTest#test_listing_subscribers:
ActionView::Template::Error: undefined method `read_multi' for nil:NilClass
Did you mean?  readline
    app/views/pages/_repos_with_pagination.html.slim:1:in `_app_views_pages__repos_with_pagination_html_slim__3459137650524765763_70239263161840'
    app/views/pages/index.html.slim:34:in `_app_views_pages_index_html_slim___1852363351166757474_70239271000460'
    test/test_helper.rb:90:in `login_via_github'
    test/integration/maintaining_repo_subscriptions_test.rb:7:in `triage_the_sandbox'
    test/integration/maintaining_repo_subscriptions_test.rb:35:in `block in <class:MaintainingRepoSubscriptionsTest>'
```

The code is assigning collection_cache store here - https://github.com/prathamesh-sonpatki/rails/blob/b165d73f2ccd421c9f3518ce9bf5bcb322440cb7/actionview/lib/action_view/railtie.rb#L42 but in this case it is nil. 

So it is failing on this line - https://github.com/prathamesh-sonpatki/rails/blob/b165d73f2ccd421c9f3518ce9bf5bcb322440cb7/actionview/lib/action_view/renderer/partial_renderer/collection_caching.rb#L16.

r? @kaspth

**Comment by kaspth** [comment]
Inspected this in my own app and there is a cache store when that line is hit. Assume CodeTriage might have a different setup?

**Comment by prathamesh-sonpatki** [comment]
@kaspth This is not happening on newly created Rails 5 app. I am investigating more where is the issue.

**Comment by sikachu** [comment]
This is weird. I can help looking into this as well.

**Comment by prathamesh-sonpatki** [comment]
@sikachu Thanks, I was not able to look into more details today. But code which fails is present here - https://github.com/codetriage/codetriage/pull/454/files

**Comment by sikachu** [comment]
After some investigation, this is actually a legit bug.

I tried to track down the location in which this cache store get introduced, and it's in AV's Railtie here: https://github.com/rails/rails/blob/v5.0.0.beta3/actionview/lib/action_view/railtie.rb#L42 (introduced in e56c6354)

I tried to put a `puts` statement there to see what `app.config.action_controller.cache_store` is. It turns out that that particular value is `nil` in test environment, but it's `ActiveSupport::Cache::NullStore` in development environment.

Digging further, I found that it might be because `cache_store` is only set in development and production environment. Consider this grep result from freshly created project:

```
$ ag cache_store -B 5
config/environments/production.rb
59-  if ENV["RAILS_LOG_TO_STDOUT"].present?
60-    config.logger = ActiveSupport::TaggedLogging.new(Logger.new(STDOUT))
61-  end
62-
63-  # Use a different cache store in production.
64:  # config.cache_store = :mem_cache_store

config/environments/development.rb
16-  if Rails.root.join('tmp/caching-dev.txt').exist?
17-    config.action_controller.perform_caching = true
18-
19-    config.action_mailer.perform_caching = false
20-
21:    config.cache_store = :memory_store
25-  else
26-    config.action_controller.perform_caching = false
27-
28-    config.action_mailer.perform_caching = false
29-
30:    config.cache_store = :null_store
```

@kaspth In this case, I think there're three ways to fix this:

1. Change that initializer to fallback to `NullStore` if `cache_store` is nil.
2. Change `action_controller.cache_store` to have `NullStore` by default.
3. Add config in `config/application.rb` for new apps to always have `NullStore` as a default storage.

What should be the preferred option? I personally think 1) would have the least side effect, but I'll let you decide.

**Comment by kaspth** [comment]
@sikachu how does that square with @prathamesh-sonpatki's comment earlier? Reposted here:

> This is not happening on newly created Rails 5 app

I also don't see it in my app I've carried through Rails 5 betas. Is it CodeTriage you're seeing the error or one you've generated?

Thanks for digging in, both of you! :heart:

**Comment by sikachu** [comment]
@kaspth that got me confused as well. Do you mind checking on your side what you see from running `rails c -e test` and `Rails.configuration.action_controller.cache_store`?

**Comment by kaspth** [comment]
Sure thing! When tracking master and beta3 I get:

```ruby
Rails.configuration.action_controller.cache_store
# => <ActiveSupport::Cache::FileStore:0x007f921cb8e240 ...>
```

**Comment by kaspth** [comment]
I'm gonna close this because I can't reproduce with a standard Rails app generated from beta3. It might be a problem with CodeTriage and one of its dependencies. Thanks for looking into this! :heart:

**Comment by lsylvester** [comment]
I have also run into this issue. It was caused by a gem that required 'action_controller'.

On start up, the  `action_view` initialisers are defined before the `action_controller` ones, so they execute before as well. If `action_controller` is already loaded, then the `on_load` callbacks run immediate, before the `action_controller` initialisers have a chance to run.

I have a fix for this in #24752
